ios/Sources/GutenbergKit/Sources/Media/MediaFileManager.swift

```
@@ -0,0 +1,160 @@
+import Foundation
+import PhotosUI
+import SwiftUI
+import UniformTypeIdentifiers
+import WebKit
+
+/// Manages media files for the editor, handling imports, storage, and cleanup.
+///
+/// Files are stored in the Library/GutenbergKit/Uploads directory and served via
+/// a custom `gbk-media-file://` URL scheme. Old files are automatically cleaned up.
+actor MediaFileManager {
+    /// Shared instance for app-wide media management
+    static let shared = MediaFileManager()
+
+    private let fileManager = FileManager.default
+
+    /// Base directory for all GutenbergKit files
+    let rootURL: URL
+
+    /// Uploads directory URL
+    private let uploadsDirectory: URL
+
+    /// Initializes the file manager with an optional custom root directory.
+    /// Creates necessary directories and schedules cleanup on initialization.
+    init(rootURL: URL? = nil) {
+        if let rootURL {
+            self.rootURL = rootURL
+        } else {
+            self.rootURL = URL.libraryDirectory.appendingPathComponent("GutenbergKit")
+        }
+        self.uploadsDirectory = self.rootURL.appendingPathComponent("Uploads")
+
+        // Create directories synchronously during init
+        try? fileManager.createDirectory(at: self.rootURL, withIntermediateDirectories: true)
+        try? fileManager.createDirectory(at: self.uploadsDirectory, withIntermediateDirectories: true)
```

If these silently fail `saveMediaData` won't work right.

Probably best to just run these on every call to `saveMediaData` if needed – makes the error handling way more straightforward.

---

Good idea; updated.